### DISCUSSION TITLE
Polling dynamic updates and layouting with handle

### DISCUSSION THREAD

**Discussion by KnisterPeter** [remix-run/remix#3136] [Q&A] [answered]
Hi :wave:,

we want to pull updates into our page rendered by remix.
The first rendering is quite fine and contains up-to-date information since we are adding all data to our `loader`. We do use the layouting mechanism and add a few components (a sidebar) into the `handle` which makes use of the loader data. This is similar to the breadcrumb example.

But now we want to update the page every few seconds with an updated server state. We do not want to reload the full page. There is the `fetcher` to make this work, but I'm unsure how to use that to update the page as well as the sidebar?
Do we need two fetchers (one in the main content area and one in the sidebar) fetching the same updates?

Any suggestion?

**Comment by sergiodxa** [comment, marked as answer]
There's a discussion to add a feature to ask Remix to refresh data #1996, in that discussion Ryan mentions some ways to achieve that right now, the best way is to use a fetcher to submit to an action returning nothing, so then Remix will trigger a revalidation of the data from the loaders.

There's an implementation of that useDataRefresh hook mentioned on [remix-utils](https://github.com/sergiodxa/remix-utils#usedatarefresh) too in case you want to try that, the internal implementation is quite simple and you could copy it too
